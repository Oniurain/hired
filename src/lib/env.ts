const DEV_MODE_IDENTIFIER = 'dev'

const BACKEND_BASE_URL = 
  `${process.env.REACT_APP_BACKEND_PROTOCOL}://${process.env.REACT_APP_BACKEND_HOST}:${process.env.REACT_APP_BACKEND_PORT}`

const backendUrl = (fragment: string): string => {
  return `${BACKEND_BASE_URL}${fragment}`
}

const isDev = (): boolean => {
  return process.env.NODE_ENV === DEV_MODE_IDENTIFIER
}

export {
  BACKEND_BASE_URL,
  DEV_MODE_IDENTIFIER,
  backendUrl,
  isDev
}
