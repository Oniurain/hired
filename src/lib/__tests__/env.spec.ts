import * as env from '../env'

describe('env', () => {
  it('should provide the correct backend-base-url', () => {
    expect(env.BACKEND_BASE_URL).toMatchSnapshot()
  })

  it('should construct a backend url', () => {
    expect(env.backendUrl('/foo/bar')).toMatchSnapshot()
  })

  it('should determine if we are in dev-mode', () => {
    expect(env.isDev()).toBe(false)
  })
})