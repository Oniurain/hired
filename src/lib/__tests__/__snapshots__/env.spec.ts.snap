// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`env should construct a backend url 1`] = `"http://localhost:8000/foo/bar"`;

exports[`env should provide the correct backend-base-url 1`] = `"http://localhost:8000"`;
